src/dialogs/quick-bar/ha-quick-bar.ts

```diff
@@ -181,16 +188,37 @@ export class QuickBar extends LitElement {
     }
 
     this._items = (this._commandMode ? this._commandItems : this._entityItems)
-      .filter(({ text, altText }) => {
-        const values = [text];
-        if (altText) {
-          values.push(altText);
-        }
-        return fuzzySequentialMatch(this._itemFilter, values);
-      })
-      .sort((itemA, itemB) => compare(itemA.text, itemB.text));
+      .filter(({ text, altText }) =>
+        altText
+          ? isPatternInWords(this._itemFilter, text, altText)
+          : isPatternInWords(this._itemFilter, text)
+      )
+      .map<ScoredItem>((item) => ({
+        item,
+        score: item.altText
+          ? fuzzySequentialMatch(this._itemFilter, item.text, item.altText)
```

Comment:
Since `fuzzySequentialMatch` also does a `isPatternInWords` we now do that twice for items that will have a match, can we filter out none matching patterns without the first `isPatternInWords `?

Reply:
Move the filter below `fuzzySequentialMatch`, and filter for `undefined` scores?

Reply:
Done with other changes